### PR TITLE
Foreground ripple support in Pressable

### PR DESCRIPTION
**Pull request by intergalacticspacehighway**
## Summary
This PR aims to enable support for foreground ripple in Pressable. This makes it possible to show ripple on top of custom child components like Image as shown in the below example.

## Changelog

[Android] [Added] - Support for foreground ripple in Pressable

## Test Plan

- Pass property useForeground: true in android_ripple config to verify the changes.

https://user-images.githubusercontent.com/23293248/120111371-4cecbf00-c18f-11eb-8acb-d10718d5483c.mov

### CODE REVIEW

**Comment by TheSavior** [comment]
Looks like the JavaScript tests are failing due to a missing dep in the useMemo array.

File: Libraries/Components/Pressable/Pressable.js [hunk 1]
```diff
@@ -180,15 +180,28 @@ function Pressable(props: Props, forwardedRef): React.Node {
       ? {...props.accessibilityState, disabled}
       : props.accessibilityState;
 
-  const restPropsWithDefaults: React.ElementConfig<typeof View> = {
+  let restPropsWithDefaults: React.ElementConfig<typeof View> = {
```
Comment (yungsters) [inline Libraries/Components/Pressable/Pressable.js:183]: Can you keep this as `const` and instead create a new `const` local variable that contains the ripple props (or remains undefined) that you spread in here?

It's a minor code reorganization, but it helps keep the code slightly more readable in the long-term the more you can have `const` variables that — as you read the declaration — you know will not be reassigned later.

File: Libraries/Components/Pressable/useAndroidRippleForView.js [hunk 2]
```diff
@@ -27,6 +27,7 @@ export type RippleConfig = {|
   color?: ColorValue,
   borderless?: boolean,
   radius?: number,
+  useForeground?: boolean,
```
Comment (yungsters) [inline Libraries/Components/Pressable/useAndroidRippleForView.js:30]: Can you just name this `foreground`? Even though it is a property name and not a variable name, avoiding the "use" prefix is just better in the current world with hooks.

File: Libraries/Components/Pressable/useAndroidRippleForView.js [hunk 3]
```diff
@@ -58,16 +61,23 @@ export default function useAndroidRippleForView(
         'Unexpected color given for Ripple color',
       );
 
+      let viewProps = {};
```
Comment (yungsters) [inline Libraries/Components/Pressable/useAndroidRippleForView.js:64]: Similar feedback as above about making this `const`. You can also instantiating this after `backgroundValue`:

```
const viewProps = foreground === true
  ? {nativeForegroundAndroid: backgroundValue}
  : {nativeBackgroundAndroid: backgroundValue};
```

Could even return this to line 80 as a ternary property initialization in the return statement.

File: Libraries/Components/Pressable/useAndroidRippleForView.js [hunk 4]
```diff
@@ -98,5 +108,5 @@ export default function useAndroidRippleForView(
       };
     }
     return null;
-  }, [color, borderless, radius, viewRef]);
+  }, [color, borderless, radius, viewRef, useForeground]);
```
Comment (yungsters) [inline Libraries/Components/Pressable/useAndroidRippleForView.js:111]: Nit: Alphabetize (after renaming to `foreground`).

**Review by yungsters — CHANGES_REQUESTED** [review]
Provided some minor feedback.

**Pull request closed without merging**